src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -640,6 +649,15 @@ private SafeCurlSlistHandle SetRequestHeaders(SafeCurlHandle handle, HttpRequest
                     rawHandle = Interop.libcurl.curl_slist_append(rawHandle, header);
                     retVal.SetHandle(rawHandle);
                 }
+
+                // Since libcurl always adds a Transfer-Encoding header, we need to explicitly block
+                // it if caller specifically does not want to set the header
+                if (request.Headers.TransferEncodingChunked.HasValue && !request.Headers.TransferEncodingChunked.Value)
+                {
+                    rawHandle = Interop.libcurl.curl_slist_append(rawHandle, NoTransferEncoding);
```

Comment:
This is not possible because of the trailing ':'. This is a format specific to libcurl and if via HttpClient, one tries to set such a header, the validation checks fail